test/cpp/rpc/test_tensorpipe_serialization.cpp

```diff
@@ -25,12 +25,20 @@ TEST(TensorpipeSerialize, Base) {
   torch::distributed::rpc::TensorPipeEntry tpEntry =
       torch::distributed::rpc::tensorpipeSerialize(sendingRpcMessage);
   tensorpipe::Message sendingTpMessage = std::move(tpEntry.message);
-  EXPECT_EQ(sendingTpMessage.tensors.size(), 2);
```

Comment:
Why removing this check? Is the number of tensors no longer 2?

Reply:
I think the main purpose of this test is that `tensorpipeAllocateMessage` is the inverse of `tensorpipeSerialize`, that is, that by passing through one and then the other one obtains the original RPC message. The implementation details of the format that these two functions choose to adopt is irrelevant to the correctness of the test, as long as they both agree on it. So for example even if the RPC message contained two tensors this doesn't have to be the case for the TP message: it could deduplicate them (if they have the same storage), coalesce them, treat one as a payload (if it's small enough that latency is more important than bandwidth), ... The test should be robust to these changes.